File: src/main/java/me/elsiff/morefish/condition/PotionEffectCondition.java
package me.elsiff.morefish.condition;

import org.bukkit.entity.Player;
import org.bukkit.potion.PotionEffectType;

public class PotionEffectCondition implements Condition {
    private PotionEffectType effectType;
    private int amplifier;

    public PotionEffectCondition(PotionEffectType effectType, int amplifier) {
        this.effectType = effectType;
        this.amplifier = amplifier;
    }

    @Override
    public boolean isSatisfying(Player player) {
        return (player.hasPotionEffect(effectType) &&
                player.getPotionEffect(effectType).getAmplifier() >= amplifier);
    }
}

File: src/main/kotlin/me/elsiff/morefish/fishing/condition/PotionEffectCondition.kt
package me.elsiff.morefish.fishing.condition

import me.elsiff.morefish.fishing.competition.FishingCompetition
import org.bukkit.entity.Item
import org.bukkit.entity.Player
import org.bukkit.potion.PotionEffectType

class PotionEffectCondition(
    private val effectType: PotionEffectType,
    private val minAmplifier: Int
) : FishCondition {
    override fun check(caught: Item, fisher: Player, competition: FishingCompetition): Boolean {
        return fisher.hasPotionEffect(effectType) && fisher.getPotionEffect(effectType)!!.amplifier >= minAmplifier
    }
}
